openmmtools/alchemy.py

```diff
@@ -1787,56 +1977,101 @@ def _alchemically_modify_NonbondedForce(self, reference_force, alchemical_region
         """
         # TODO Change softcore_beta to a dimensionless scalar to multiply some intrinsic length-scale, like Lennard-Jones alpha.
         # TODO Try using a single, common "reff" effective softcore distance for both Lennard-Jones and Coulomb.
-
+        forces_by_lambda = {}
```

Comment:
It may be good for readability to initialize the `forces_by_lambda` variable right before you use it at the bottom of the function.